src/command.ts

```diff
@@ -27,22 +30,55 @@ export class Command {
             },
         };
 
-        const exitCode = await exec.exec(executable, args, finalExecOptions);
+        if (execOptions.group) {
+            const groupName = [ executable, ...args ].join(" ");
+            core.startGroup(groupName);
+        }
+
+        try {
+            const exitCode = await exec.exec(executable, args, finalExecOptions);
 
-        if (execOptions.ignoreReturnCode !== true && exitCode !== 0) {
-            // Throwing the stderr as part of the Error makes the stderr show up in the action outline,
-            // which saves some clicking when debugging.
-            let error = `${path.basename(executable)} exited with code ${exitCode}`;
-            if (stderr) {
-                error += `\n${stderr}`;
+            if (execOptions.ignoreReturnCode !== true && exitCode !== 0) {
+                // Throwing the stderr as part of the Error makes the stderr show up in the action outline,
+                // which saves some clicking when debugging.
+                let error = `${path.basename(executable)} exited with code ${exitCode}`;
+                if (stderr) {
+                    error += `\n${stderr}`;
+                }
+                throw new Error(error);
             }
-            throw new Error(error);
+
+            return {
+                exitCode, output: stdout, error: stderr,
+            };
         }
 
-        return {
-            exitCode,
-            stdout,
-            stderr,
-        };
+        finally {
+            if (execOptions.group) {
+                core.endGroup();
+            }
+        }
+    }
+
+    public static async tag(image: string, tags: string[]): Promise<void> {
+        // get docker cli
+        let dockerPath: string;
+
+        try {
+            dockerPath = await io.which("docker", true);
+        }
+        catch (error) {
+            core.debug(error);
+            core.setFailed(
+                "❌ Docker client not found. Make sure that docker client is installed before running this action"
+            );
+            return;
+        }
+        const args: string[] = [ "tag" ];
+        for (const tag of tags) {
+            args.push(`${image}:${tag}`);
+        }
+        core.info(`Tagging the built image with tags ${tags.toString()}`);
```

Comment:
can you use `.join` instead of `.toString` on arrays for better output

```
> const tags = [ "v1", "v2", "v3" ]
undefined
> tags.toString()
'v1,v2,v3'
> tags.join(", ")
'v1, v2, v3'
```

or use https://github.com/redhat-actions/openshift-actions-runner-installer/blob/main/src/util/util.ts#L12